onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1407,6 +1407,369 @@ with the exception that numpy default keepdims to False instead of True.)DOC")
           "Keep the reduced dimension or not, default 1 mean keep reduced dimension.",
           AttributeProto::INT);
 
+  ONNX_CONTRIB_OPERATOR_SCHEMA(QLinearAdd)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetDoc(R"DOC(Performs element-wise binary quantized addition (with Numpy-style broadcasting support).
+This operator supports **multidirectional (i.e., Numpy-style) broadcasting**.
+```
+C = (A_scale * (A - A_zero_point) + B_scale * (B - B_zero_point))/C_scale + C_zero_point
+```
+)DOC")
+      .Input(0, "A", "First operand.", "T")
+      .Input(
+          1,
+          "A_scale",
+          "Input A's scale. It's a scalar, which means a per-tensor/layer quantization.",
+          "tensor(float)")
+      .Input(
+          2,
+          "A_zero_point",
+          "Input A zero point. Default value is 0 if it's not specified. It's a scalar, which means a per-tensor/layer quantization.",
+          "T",
+          OpSchema::Optional)
+      .Input(3, "B", "Second operand.", "T")
+      .Input(
+          4,
+          "B_scale",
+          "Input B's scale. It's a scalar, which means a per-tensor/layer quantization.",
+          "tensor(float)")
+      .Input(
+          5,
+          "B_zero_point",
+          "Input B zero point. Default value is 0 if it's not specified. It's a scalar, which means a per-tensor/layer quantization.",
+          "T",
+          OpSchema::Optional)
+      .Input(
+          6,
+          "C_scale",
+          "Output scale. It's a scalar, which means a per-tensor/layer quantization.",
+          "tensor(float)")
```

Comment:
DynamicQuantizeLinear is a function in ONNX which fuses Dynamic scale computation + dynamic zero point computation + QuantizeLinear .... This will be used when an operator input needs to be quantized to 8 bits.

The presence of this operator does not mean we will not add any operators which have input and output as 8 bits. QLinearAdd is similar to QLinearMatMul and QLinearConv where both inputs and outputs are 8 bits...

Reply:
That makes sense. My comment was just concerning models with the calling pattern of DequantizeX2->Add->DynamicQuantizeLinear. If this operator could output float then that calling pattern could by QLinearAdd->DynamicQuantizeLinear.